### ISSUE TITLE
Cannot use post-related Conditions for Reactions on PHP 5.2

### ISSUE THREAD

**Issue by JDGrimes** [WordPoints/wordpoints#654]
When attempting to save a reaction with a condition on the post title, for example, everything works fine on newer versions of PHP. But when testing on PHP 5.2, there is an error:

>The &#8220;comment\\post&#8221; arg is not registered for this event.

![wordpoints_ _points_types_ _wp_5 2_ _wordpress_-_2017-05-10_09 48 01](https://cloud.githubusercontent.com/assets/4005415/25901879/0ad91df2-3566-11e7-85e3-45f9a3406409.png)

**Comment by JDGrimes** [comment]
Note how there are two backslashes in the error, rather than just one. This indicated to me that probably that was a part of the issue. Upon further investigation, it turns out that it is. The second backslash is there for escaping, but we'd generally expect all escaping backslashes to be removed by `wp_unslash()`, which is passed over the data sent to the server. I checked to see if we were double-slashing *before* sending the data to the server, and it turns out that we are not, the value sent from our JS only has one slash. So when the data is received on the server side, that is when the escaping slashes are being added.

Of course, we know that WordPress emulates the `magic_quotes_gpc` config value, however, that is supposed to be reversed by running the slashed data through `wp_unslash()`, as I said above.

In this case, however, the slashing isn't provided by WordPress's emulating of magic quotes, but by the magic quotes setting built into PHP, which is on by default on older versions like PHP 5.2.

Apparently there is a difference in behavior between WordPress's magic slashing, and PHP's built-in magic slashing. As [noted in the manual](https://secure.php.net/manual/en/security.magicquotes.disabling.php#69024), PHP's magic quote functionality also slashes *keys*, and not just values, whereas most emulations of it only slash the values (as is true of WordPress). So WordPress also fails to unslash the keys in `wp_unslash()`. As a result, the key is slashed by PHP, and not unslashed by WordPress, leaving the extra slash and making the value invalid.

Having searched Trac for any signs of talk of this issue, it appears that it has never been brought up before. I'm thinking of opening a ticket, since the bug is really there, but by now code is probably expecting the aberrant behavior. What tangled webs we weave...

**Comment by JDGrimes** [comment]
[WP#40716](https://core.trac.wordpress.org/ticket/40716)

**Comment by JDGrimes** [comment]
Note that there was also [a bug in older versions of PHP 5.2](https://bugs.php.net/bug.php?id=41093), causing top-level keys to not always be slashed.

**Comment by JDGrimes** [comment]
Because this is not a critical bug, and only limits the ability to use conditions on post-related args on old versions of PHP that have `magic_quotes_gpc` enabled, I'm leaning toward not fixing it. Because [magic quotes were removed from PHP in 5.4](https://secure.php.net/manual/en/security.magicquotes.php), this only affects 5.2 and 5.3 (I've tested on both). And because it was deprecated in 5.3, it defaults to off even then. Those versions collectively make up only 18% of WordPress sites, and even less on the latest versions of WordPress that WordPoints supports. 

![about_ _statistics_ _wordpress_-_2017-05-10_15 16 32](https://cloud.githubusercontent.com/assets/4005415/25916760/cf22d8b0-3593-11e7-8415-8f8f38a89570.png)

Because it mainly affects 5.2, the number of affected sites for us is likely somewhere like 3-4%.

Since we intend to drop PHP 5.2 soon (and PHP 5.3 might be soon to follow), I don't know that it is worth fixing this on our end. If WordPress decides to fix it though, that's great.
